src/pkgcheck/checks/python.py

```diff
@@ -694,3 +694,39 @@ def feed(self, pkg):
                 if GITHUB_ARCHIVE_RE.match(uri):
                     yield PythonGHDistfileSuffix(f.filename, uri, pkg=pkg)
                     break
+
+
+class PythonMismatchedPackageName(results.PackageResult, results.Info):
+    """Package name does not following PyPI-based naming policy"""
+
+    def __init__(self, expected, **kwargs):
```

Comment:
Slowly I'm trying to include typing where possible

```suggestion
    def __init__(self, expected: str, **kwargs):
```